test/machineslib.py

```diff
@@ -42,21 +42,18 @@ def waitPageInit(self):
         with self.browser.wait_timeout(30):
             self.browser.wait_in_text("body", "Virtual machines")
 
-    def performAction(self, vmName, action, checkExpectedState=True, connectionName="system"):
+    def performAction(self, vmName, action, checkExpectedState=True, connectionName="system", logPath=None):
         b = self.browser
         m = self.machine
 
-        def getStartTime():
-            return m.execute(f"grep 'starting up' /var/log/libvirt/qemu/{vmName}.log | tail -1")
-
-        initialTime = getStartTime()
         b.click(f"#vm-{vmName}-{connectionName}-action-kebab button")
         b.wait_visible(f"#vm-{vmName}-{connectionName}-action-kebab > .pf-v5-c-dropdown__menu")
         b.click(f"#vm-{vmName}-{connectionName}-{action} a")
-        if (action == "reboot" or "forceReboot") and connectionName == "system":
+        if (action == "reboot" or "forceReboot") and connectionName == "system" and logPath:
             # Check VM doesn't get rebooted after opening dialog
             # https://bugzilla.redhat.com/show_bug.cgi?id=2221144
-            self.assertEqual(initialTime, getStartTime())
+            # Need to clear the log before invoking
+            testlib.wait(lambda: "Linux version" not in m.execute(f"cat {logPath}"))
```

Comment:
The comment here is misleading -- it sounds like the code would clear the log, but it doesn't -- it expects that the log gets cleared through the reboot. If you don't mind, I'll change this, and also flesh out the commit message.

Reply:
Also, the whole comment above is also bogus/outdated -- at this point the click already happened, so this isn't "after opening the dialog (what bz#2221144 was about), but after the action.